RFCs/FS-1004-result-type.md

```diff
@@ -48,3 +48,42 @@ The standard range of testing considerations for F# library types apply.
 - declare as struct
 - compiled name
 - require additional functions?
+
+### Proposal for additional functions
+
+Minimal:
+
+```fsharp
+bind : ('TSuccess -> Result<'T, 'TError>) -> Result<'TSuccess, 'TError> -> Result<'T, 'TError>
+map : ('TSuccess -> 'T) -> Result<'TSuccess, 'TError> -> Result<'T, 'TError>
+```
+
+Nice-to-have:
+
+```fsharp
+fold : ('TSuccess -> 'T) -> ('TError -> 'T) -> Result<'TSuccess, 'TError> -> 'T
+mapError : ('TError -> 'T) -> Result<'TSuccess, 'TError> -> Result<'TSuccess, 'T>
+bimap : ('TSuccess -> 'T) -> ('TError -> 'U) -> Result<'TSuccess, 'TError> -> Result<'T, 'U>
+// `bimap` or `mapBoth`?
+```
+
+Low priority -- should we consider the following to get some parity with the `Option` module?
+
+```fsharp
+iter : ('TSuccess -> unit) -> Result<'TSuccess, 'TError> -> unit
```

Comment:
I don't think ``iter`` adds much over an explicit match. It is also "dangerous" as it drops information if a failure occurred.